### ISSUE TITLE
Support standard input in `IntervalListTools`

### ISSUE THREAD

**Issue by nh13** [broadinstitute/picard#1551]
I would like to use `IntervalListTools` with standard input. 
Right now I get:

```
Exception in thread "main" htsjdk.samtools.SAMException: Cannot figure out type of file /dev/stdin from extension. Current implementation understands the following types: [VCF: [.bcf, .vcf.gz, .vcf], INTERVAL_LIST: [.interval_list]]
```

I would either like the input to be assumed to be a `.interval_list`, or be able to directly specify the type.

**Comment by yfarjoun** [comment]
that is indeed annoying. it would be nice to have a smart system like GATK where it reads the header and tries to figure it out on it's own....not by the extension. perhaps the GATK smartness can be pushed down to htsjdk? @lbergelson ?

**Comment by lbergelson** [comment]
I think pushing down the gatk feature reading system would be a good idea, although not a trivial one.  In this case it's not going to help too much because the IntervalListCodec only checks the file extension.  It's tricky for interval files in general because they don't all have an unambiguous header to check.  We generally need a mechanism for explicitly specifying the file type on load.  Some gatk tools use an awkward syntax like --Input:VCF my_path_with_no_extension but it's a bit of a weird solution and not universally enabled even in GATK.  ReadBundles are sidestepping it by adding a type field to their json format alongside the path but that's also awkward.  If you're reading from stdin you probably don't want to have to pass in a json file to specify it.  

It might be a good idea to just add a default in this case.

**Comment by nh13** [comment]
I am in favor for this tool a default to the interval list type with a warning message that it's defaulting to that type.  It's called `IntervalListTools` after all :P

**Comment by yfarjoun** [comment]
:+1: seems like a simple change...replace the `throw`
in picard/util/IntervalListTools.java:599 with a warning and return
a picard.util.IntervalListTools.IntervalListInputType#INTERVAL_LIST##
getIntervalListInternal.

only problem is that the method is currently protected....

**Comment by yfarjoun** [comment]
fixed by #1582
